L1Trigger/Phase2L1Taus/plugins/L1HPSPFTauProducer.h

```diff
@@ -0,0 +1,71 @@
+#ifndef L1Trigger_Phase2L1Taus_L1HPSPFTauProducer_h
+#define L1Trigger_Phase2L1Taus_L1HPSPFTauProducer_h
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "L1Trigger/Phase2L1Taus/interface/L1HPSPFTauQualityCut.h"  // L1HPSPFTauQualityCut
+#include "L1Trigger/Phase2L1Taus/interface/L1HPSPFTauBuilder.h"     // L1HPSPFTauBuilder
+#include "DataFormats/Phase2L1Taus/interface/L1HPSPFTau.h"          // l1t::L1HPSPFTau
+#include "DataFormats/Phase2L1Taus/interface/L1HPSPFTauFwd.h"       // l1t::L1HPSPFTauCollection
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"  // l1t::PFCandidate, l1t::PFCandidateCollection, l1t::PFCandidateRef
+#include "DataFormats/JetReco/interface/CaloJet.h"
+#include "DataFormats/L1TCorrelator/interface/TkPrimaryVertex.h"
+
+#include <string>
+#include <vector>
+
+class L1HPSPFTauProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1HPSPFTauProducer(const edm::ParameterSet& cfg);
+  ~L1HPSPFTauProducer() = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event& evt, const edm::EventSetup& es);
+
+  std::string moduleLabel_;
+
+  std::unique_ptr<L1HPSPFTauBuilder> tauBuilder_;
```

Comment:
from reading the code, I think that `tauBuilder_` could simply be a data member, instead of a pointer ?
*i.e.*
```suggestion
  L1HPSPFTauBuilder tauBuilder_;
```

construct it directly in the member initialiser list like `tauBuilder_(cfg),`

and use it by value (`tauBuilder_.method()`) instead of by pointer (`tauBuilder_->method()`).